mysql-test/suite/rpl/t/rpl_using_gtid_default.test

```diff
@@ -0,0 +1,170 @@
+#
+# Purpose:
+#   This test ensures that a replica's default value for Using_Gtid is set
+# correctly. Specifically, it should default to 'Slave_Pos' unless the primary
+# server does not support GTIDs (if its version is less than 10), in which case
+# the replica should fall back to 'No'.
+#
+# Methodology:
+#   Validate the value of Using_Gtid on replica initialization and after
+# RESET SLAVE commands. Specifically, we validate the following use cases:
+#
+#   Case 1) A replica will initialize with Slave_Pos if the primary supports
+#           GTIDs
+#   Case 2) A replica configured with the Using_Gtid=Slave_Pos issued
+#           RESET SLAVE will preserve Using_Gtid without any informational
+#           messages.
+#   Case 3) A replica configured with Using_Gtid=No against a master which
+#           supports GTIDs will revert to Using_Gtid=Slave_Pos after issued
+#           RESET SLAVE and provide an informational note
+#   Case 4) A fresh replica targeting a primary which does not support GTIDs
+#           will fall back to Using_Gtid=No when starting. An informational
+#           message should be logged.
+#   Case 5) A replica connected to a primary which does not support GTIDs
+#           should preserve Using_Gtid=No when issued RESET SLAVE. No message
+#           should be provided to the user.
```

Comment:
I like `Using_Gtid=No` preservation indent. Still [`RESET SLAVE ALL`](https://mariadb.com/kb/en/reset-replica) forgets everything about the last master.